dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+namespace Microsoft.Windows.PushNotifications
+{
+    // Event args for the Push payload.
+    runtimeclass PushNotificationReceivedEventArgs
+    {
+        // Initialize using the IBackgroundInstance: used specifically for the Background Activation scenario
+        static PushNotificationReceivedEventArgs CreateFromBackgroundTaskInstance(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+// Initialize using the PushNotificationEventArgs from Windows: used specifically for in-memory event handling when app is already in foreground
+static PushNotificationReceivedEventArgs CreateFromPushNotificationReceivedEventArgs(Windows.Networking.PushNotifications.PushNotificationReceivedEventArgs args);
+
+// The Push payload
+byte[] Payload { get; };
+
+// Gets a deferral to run under specific modes like low power mode
+Windows.ApplicationModel.Background.BackgroundTaskDeferral GetDeferral();
+
+// Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+// Set to true to prevent proceeding launch due to Background Activation: false by default
+Boolean Handled;
+};
+
+[flags]
+enum PushNotificationRegistrationKind
+{
+    PushTrigger = 0x1, // Registers a Push Trigger with Background Infrastructure
+    ComActivator = 0x2, // Registers the Project Reunion Background Task component as an InProc COM server 
+};
+
+// An abstraction over the activation Registration flow
```

Comment:
Comment is confusing to those outside the OS. We are abstracting an OS thing, but outside developers don't see the OS thing, do they?

Reply:
I don't see value in that comment. Can we remove it? @pmpurifoy